crates/rspack_core/src/module_graph.rs

```diff
@@ -12,8 +11,7 @@ use crate::{
   BoxModule, BoxModuleDependency, DependencyId, Module, ModuleGraphModule, ModuleIdentifier,
 };
 
-// FIXME: placing this as global id is not acceptable, move it to somewhere else later
-static NEXT_MODULE_GRAPH_CONNECTION_ID: AtomicUsize = AtomicUsize::new(1);
+type ModuleGraphConnectionId = usize;
```

Comment:
These ids should be wrapped inside as newtype, but I'll leave it to the next person ;-)